beacon_node/network/src/network_beacon_processor/rpc_methods.rs

```diff
@@ -428,6 +429,105 @@ impl<T: BeaconChainTypes> NetworkBeaconProcessor<T> {
         Ok(())
     }
 
+    pub fn handle_light_client_updates_by_range(
+        self: &Arc<Self>,
+        peer_id: PeerId,
+        connection_id: ConnectionId,
+        substream_id: SubstreamId,
+        request_id: RequestId,
+        request: LightClientUpdatesByRangeRequest,
+    ) {
+        self.terminate_response_stream(
+            peer_id,
+            connection_id,
+            substream_id,
+            request_id,
+            self.clone()
+                .handle_light_client_updates_by_range_request_inner(
+                    peer_id,
+                    connection_id,
+                    substream_id,
+                    request_id,
+                    request,
+                ),
+            Response::LightClientUpdatesByRange,
+        );
+    }
+
+    /// Handle a `LightClientUpdatesByRange` request from the peer.
+    pub fn handle_light_client_updates_by_range_request_inner(
+        self: Arc<Self>,
+        peer_id: PeerId,
+        connection_id: ConnectionId,
+        substream_id: SubstreamId,
+        request_id: RequestId,
+        req: LightClientUpdatesByRangeRequest,
+    ) -> Result<(), (RpcErrorResponse, &'static str)> {
+        debug!(self.log, "Received LightClientUpdatesByRange Request";
+            "peer_id" => %peer_id,
+            "count" => req.count,
+            "start_period" => req.start_period,
+        );
+
+        // Should not send more than max light client updates
+        let max_request_size: u64 = req.max_requested();
+        if req.count > max_request_size {
+            return Err((
+                RpcErrorResponse::InvalidRequest,
+                "Request exceeded max size",
+            ));
+        }
+
+        let lc_updates = match self
+            .chain
+            .get_light_client_updates(req.start_period, req.count)
+        {
+            Ok(lc_updates) => lc_updates,
+            Err(e) => {
+                error!(self.log, "Unable to obtain light client updates";
```

Comment:
This error can happen if we just don't have the old updates available, right? Maybe this log should be debug level

Reply:
Calls to `iter_column_from` can error if there's no data on the range? Or based on the split position?

Reply:
Calls to `iter_column_from` shouldn't error if there's no data. i believe it'd just return an empty iterator in that case.  the error cases here are
- ssz decoding failures
- `remove_column_variable` error

so i think an error log in this case makes sense as it means theres something seriously wrong with the light client data stored in the db

Reply:
ah right, didn't look under the hood at the `iter_column_from`

this is fine then